solr/core/src/test/org/apache/solr/cloud/LeaderElectionTest.java

```diff
@@ -258,8 +258,8 @@ public void testCancelElection() throws Exception {
     Thread.sleep(1000);
 
     String urlScheme = zkStateReader.getClusterProperty(URL_SCHEME, "http");
-    String url1 = Utils.getBaseUrlForNodeName("127.0.0.1:80_solr/1", urlScheme) + "/";
-    String url2 = Utils.getBaseUrlForNodeName("127.0.0.1:80_solr/2", urlScheme) + "/";
+    String url1 = Utils.getBaseUrlForNodeName("127.0.0.1:80_solr/1", urlScheme) + "/" + "1" + '/';
```

Comment:
I have some concerns about how this test works....   I kind of worked around it, however the naming of the urls for leader election seemed "odd".

Reply:
So, I actually am worried this entire idea may fail..  At least the way I read this test, when we do leaderelection, we name our nodes `{some_ip}/solr/1` and `{same_ip}/solr/2` and `{same_ip}/solr/3` and then something happens.   However, I don't know if that is ACTUALLY how it happens, or maybe just how this test was written, since I've never dug into leader election in any way....    I also was wondering if maybe we need some sort of `.bats` test of leader election to prove that this change didn't break leader election in the real world..  I mean, technically this LeaderElectionTest.java passes...  but...

Reply:
@psalagnac WDYT of the changes to this test?

Reply:
Thanks for tagging @psalagnac...   I think getting this resolved is the last thing that needs doing...  I am wondering if I need to create some sort of `.bats` test that validates the happy path of leaderelection...?

Reply:
Thanks for pining me on this.

This is related to other David's  [comment](https://github.com/apache/solr/pull/1810/files/49fe85375b74cd76d7bc08bcc0d32781632bc0d6#r1279827024) on the path removal in `Utils.getBaseUrlForNodeName`.

The change in the test does not make sense to me since the core name is here twice, but it is only once in the result URL. We just want to build the core url from the node base URL. I think it should be like this:

```
    String url1 = Utils.getBaseUrlForNodeName("127.0.0.1:80_solr", urlScheme) + "/1/";
```